controllers/feeds/getAnonymousFeedChat.js

```diff
@@ -14,15 +14,24 @@ const {
 
 const getAnonymousFeedChatService = async (req, res) => {
   try {
+    let {last_fetch_date = null} = req.query;
+    if (last_fetch_date) {
+      last_fetch_date = new Date(last_fetch_date);
+      last_fetch_date = last_fetch_date.toISOString();
+    }
     const mySignedId = await UsersFunction.findSignedUserId(User, req.userId);
     const data = await getstreamService.notificationGetNewFeed(req.userId, req.token);
     const newFeed = [];
 
     // eslint-disable-next-line no-restricted-syntax
     for (const feeds of data.results) {
+      if (last_fetch_date && feeds.updated_at < last_fetch_date) {
+        continue;
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This filtering logic could be improved for performance by using the `filter()` method on the `data.results` array instead of using a `for...of` loop and `continue`. This would make the code more readable and efficient.
```diff
-    // eslint-disable-next-line no-restricted-syntax
-    for (const feeds of data.results) {
-      if (last_fetch_date && feeds.updated_at < last_fetch_date) {
-        continue;
-      }
-      const mapping = mappingFeed(req, feeds);
-      newFeed.push(...mapping);
-    }
+    const filteredFeeds = data.results.filter(feed => !last_fetch_date || feed.updated_at >= last_fetch_date);
+    for (const feed of filteredFeeds) {
+      const mapping = mappingFeed(req, feed);
+      newFeed.push(...mapping);
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->